exoscale/api/compute.py

```diff
@@ -0,0 +1,2088 @@
+            raise APIException(e.error["errortext"], e.error)
+
+    def unset_reverse_dns(self):
+        """
+        Unset the public network interface IP address reverse DNS record.
+
+        Returns:
+            None
+        """
+
+        try:
+            self.compute.cs.deleteReverseDnsFromVirtualMachine(id=self.id)
+        except CloudStackApiException as e:
+            raise APIException(e.error["errortext"], e.error)
+
+    def delete(self):
+        """
+        Delete the instance.
+
+        Returns:
+            None
+        """
+
+        try:
+            self.compute.cs.destroyVirtualMachine(id=self.id)
+        except CloudStackApiException as e:
+            raise APIException(e.error["errortext"], e.error)
+
+        self.compute = None
+        self.res = None
+        self.id = None
+        self.name = None
+        self.zone = None
+        self.type = None
+        self.template = None
+        self.ipv4_address = None
+        self.ipv6_address = None
+        self.ssh_key = None
+
+    def _default_nic(self, nics):
+        for nic in nics:
+            if nic["isdefault"]:
+                return nic
+
+
+@attr.s
+class InstanceTemplate(Resource):
+    """
+    A Compute instance template.
+
+    Attributes:
+        id (str): the template unique identifier
+        name (str): the template name
+        description (str): the template description
+        zone (Zone): the zone in which the template is located
+        date (datetime.datetime): the template creation date
+        size (int): the template disk size
+        ssh_key_enabled (bool): a flag indicating whether the SSH key deployment is
+            enabled
+        password_reset_enabled (bool): a flag indicating whether the user password can
+            be reset
+
+    References:
+        * `Python datetime module`_
+
+        .. _Python datetime module: https://docs.python.org/3/library/datetime.html
+    """
+
+    compute = attr.ib(repr=False)
+    res = attr.ib(repr=False)
+    id = attr.ib()
+    name = attr.ib()
+    description = attr.ib(repr=False)
+    zone = attr.ib(repr=False)
+    date = attr.ib(repr=False)
+    size = attr.ib(repr=False)
+    ssh_key_enabled = attr.ib(default=True, repr=False)
+    password_reset_enabled = attr.ib(default=True, repr=False)
+    username = attr.ib(default=None, repr=False)
+
+    @classmethod
+    def from_cs(cls, compute, res):
+        return cls(
+            compute,
+            res,
+            id=res["id"],
+            name=res["name"],
+            description=res.get("displaytext", ""),
+            zone=compute.get_zone(id=res["zoneid"]),
+            date=datetime.strptime(res["created"], "%Y-%m-%dT%H:%M:%S%z"),
+            size=res["size"],
+            username=res["details"].get("username", None),
+            ssh_key_enabled=res["sshkeyenabled"],
+            password_reset_enabled=res["passwordenabled"],
+        )
+
+    # TODO: @classmethod def register(...)
```

Comment:
Registration/deletion are left for later?

Reply:
Yep that's procrastination from me, I kept pushing this one further later but I should now get to it now 😏 BTW this call will not be able to be tested similarly to other calls, given that we need an image to register and I'm not confident to reference an external file in stored SOS that can be deleted by accident...

Reply:
Implemented in cf74ae6 and 838c413.